src/librustc_mir/transform/check_unsafety.rs

```diff
@@ -537,7 +537,9 @@ fn unsafety_check_result(tcx: TyCtxt<'_>, def_id: DefId) -> UnsafetyCheckResult
     let mut checker = UnsafetyChecker::new(
         const_context, min_const_fn,
         body, source_scope_local_data, tcx, param_env);
-    checker.visit_body(body);
+    let mut cache = body.cache().clone();
+    let read_only_cache = ReadOnlyBodyCache::from_external_cache(&mut cache, body);
```

Comment:
why do you need a mutable cache? doesn't `mir_built` always return an already generated cache so you can just unwrap it?

Reply:
`mir_built` currently doesn't ensure the cache is generated. Should it? I hadn't done it because I saw that the `mir_const` fn would immediately run passes on it and invalidate it.